Dockerfile

```diff
@@ -1,34 +1,57 @@
-#
+### ### ### ### ### ### ### ### ###
 # Builder layer
-#
+
 FROM node:6-alpine as ghost-builder
-RUN npm install --loglevel=error -g ghost-cli
 
-ENV GHOST_VERSION 1.17.1
-ENV GHOST_INSTALL /var/lib/ghost
-ENV GHOST_CONTENT /var/lib/ghost/content
-ENV GHOST_USER node
+RUN \
+    apk update && apk upgrade                       	&& \
+    apk add --no-cache wget unzip ca-certificates   	&& \
+    echo												&& \
+    echo "--- Install ghost-cli --- "; echo         	&& \
+    npm install --loglevel=error -g ghost-cli			;
+
+ENV GHOST_VERSION="1.17.1"							    \
+	GHOST_INSTALL="/var/lib/ghost"					    \
+	GHOST_CONTENT="/var/lib/ghost/content"			    \
+	GHOST_THEMES="/var/lib/ghost/content/themes"	    \
+	GHOST_USER="node"
 
+# Set default directory
 WORKDIR $GHOST_INSTALL
 
 # Run SQLite as database
 RUN \
-	ghost install "$GHOST_VERSION" --db sqlite3 --no-prompt --no-stack --no-setup --dir "$GHOST_INSTALL"; \
-	ghost config --ip 0.0.0.0 --port 2368 --no-prompt --db sqlite3 --url http://localhost:2368 --dbpath "$GHOST_CONTENT/data/ghost.db"; \
-	ghost config paths.contentPath "$GHOST_CONTENT";
+	ghost install "$GHOST_VERSION" \
+		--db sqlite3 --no-prompt \
+		--no-stack --no-setup \
+		--dir "$GHOST_INSTALL"							&& \
+	ghost config --ip 0.0.0.0 \
+		--port 2368 --no-prompt --db sqlite3 \
+		--url http://localhost:2368 \
+		--dbpath "$GHOST_CONTENT/data/ghost.db"			&& \
+	ghost config paths.contentPath "$GHOST_CONTENT"		;
 
 COPY run-ghost.sh $GHOST_INSTALL
 
-#
+# Keeping Original GhostContent to be copied into the mounted volume (if empty)
+RUN cp -r "$GHOST_CONTENT" "$GHOST_INSTALL/content.bck"	;
+
+
+### ### ### ### ### ### ### ### ###
 # Final image
-#
+# No tzdata as it's not working on alpine3.4 (from node6)
+
 FROM node:6-alpine
 LABEL maintainer="Marco Mornati <marco@mornati.net>"
 
-ENV GHOST_VERSION 1.17.1
-ENV GHOST_INSTALL /var/lib/ghost
-ENV GHOST_CONTENT /var/lib/ghost/content
-ENV GHOST_USER node
+RUN apk update && apk upgrade		                && \
+	apk add --no-cache tini			                && \
+	rm -rf /var/cache/apk/*			                ;
```

Comment:
Why you are not executing it with no cache instead of removing it later?
What is the tini for?

Reply:
As I said earlier:

tini

I tried to use the ENTRYPOINT with run-ghost.sh but I was getting permission issues. It would be better to run this way to we could run CMD [ "/sbin/tini", "--", "/bin/sh", "-c", "node", "current/index.js" ]. Ghost would run on PID 1. It's really about best practice.

Reply:
https://github.com/krallin/tini#why-tini

Reply:
Good. I didn't know. Thanks for the information.